noise/README.md

```diff
@@ -210,27 +210,30 @@ The Noise Protocol Framework caters for sending early data alongside handshake
 messages. We leverage this construct to transmit:
 
 1. the libp2p identity key along with a signature, to authenticate each party to
-   the other.
-2. arbitrary data private to the libp2p stack. This facility is not exposed to
-   userland. Examples of usage include streamlining muxer selection.
-
-These payloads MUST be inserted into the first message of the handshake pattern
```

Comment:
I don't think we want to remove this. If anything we should define a new payload type that is **only** sent on the first message. I think it's a footgun to have the `libp2p Handshake Payload` be sent either in plaintext or encrypted. 

I also think with the current usecases we don't need to define the type of data for the first handshake message. Everything we want to do can be done with the 2nd and 3rd message. So we can keep our original crypto guarantees.

Reply:
I reverted this change.

Reply:
I rewrote this paragraph, to clarify that extensions are sent in message 2 and 3.